python/tests/compat/test_mantid.py

```diff
@@ -591,6 +591,23 @@ def test_validate_units_throws(self):
             with self.assertRaises(RuntimeError):
                 mantidcompat.validate_dim_and_get_mantid_string(i)
 
+    def test_WorkspaceGroup_parsed_correctly(self):
+        from mantid.simpleapi import (mtd, CreateSampleWorkspace,
+                                      GroupWorkspaces)
+        CreateSampleWorkspace(OutputWorkspace="ws1")
+        CreateSampleWorkspace(OutputWorkspace="ws2")
+        CreateSampleWorkspace(OutputWorkspace="ws3")
+        GroupWorkspaces(InputWorkspaces="ws1,ws2,ws3",
+                        OutputWorkspace="NewGroup")
```

Comment:
The above would still work ok if `StoreInADS=False`?

Reply:
Not in the way it is currently setup. The algorithm GroupWorkspaces takes a list of strings as inputs so can only construct a group from workspaces which already exist in the ADS. Likewise the add method on `WorkspaceGroup` takes a name string. I think this is due to the fact that WorkspaceGroups don't actually own there sub-workspaces but instead just store the names these workspaces have in the ADS.

Reply:
Ok, thanks for reminding me. So if a workspace group exists, the names also should exist. That should be fine then.